src/java.base/linux/classes/sun/nio/fs/LinuxWatchService.java

```diff
@@ -296,9 +301,12 @@ void implCloseAll() {
 
             // free resources
             unsafe.freeMemory(address);
-            UnixNativeDispatcher.close(socketpair[0]);
-            UnixNativeDispatcher.close(socketpair[1]);
-            UnixNativeDispatcher.close(ifd);
+            try {
+                UnixNativeDispatcher.close(socketpair[0]);
+                UnixNativeDispatcher.close(socketpair[1]);
+                UnixNativeDispatcher.close(ifd);
+            } catch (UnixException ignore) {
```

Comment:
Hello Brian,
Do you think it would be better for each of these `close` calls to have a try/catch/ignore block of its own? That way, for example, if the close for `socketpair[0]` fails, then we at least attempt closing the other open ones (`socketpair[1]` and `ifd`)?